pkgs/servers/seafile-server/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, lib, fetchFromGitHub, pkg-config, python3Packages, autoreconfHook
+, libuuid, sqlite, glib, libevent, libsearpc, openssl, fuse, libarchive, which
+, vala, cmake, libevhtp, oniguruma }:
+
+stdenv.mkDerivation rec {
+  pname = "seafile-server";
+  version = "8.0.4";
+
+  src = fetchFromGitHub {
+    owner = "haiwen";
+    repo = "seafile-server";
+    rev = "v${version}-server";
+    sha256 = "1lmp3hld3sgfjbzzv0vigpfgyrx37277wwdg2bvm5ipl9as2jga8";
+  };
+
+  patches = [ ./recent-libevhtp.patch ];
```

Comment:
I'm not sure this is enough, in my most recent test http upload failed.

Reply:
@schmittlauch I've reverted to using the seafile fork.
Relying on an up-to-date libevhtp would be nice though, could this patch be upstreamed?

Reply:
Oh damn, so maybe some parts of the patch are wrong or missing.
Looking at the roadmap, the next major release might finally drop libevhtp by rewriting the file server component in Go. Until then we either need to fix the patch or ship ancient code with known vulnerabilities (like upstream does :shrug: )